src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -29,16 +31,39 @@ public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorksp
             // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
             var extensionManager = workspace.GetVsService<SVsExtensionManager, IVsExtensionManager>();
 
-            var builder = ImmutableArray.CreateBuilder<HostDiagnosticAnalyzerPackage>();
-            foreach (var extension in extensionManager.GetEnabledExtensions(AnalyzerContentTypeName))
+            // get rootfolder and shellfolder location
+            string rootFolder;
+            string shellFolder;
+            if (TryGetRootAndShellFolder(extensionManager, out shellFolder, out rootFolder))
```

Comment:
@heejaechang @mavasani The new shell API seems straightforward to expose, doesn't it? If there is a good chance of getting the new API, I don't mind the temporary duplication. However, if we don't get the new API, then I would begin to worry about the duplication...